lib/Sema/CSGen.cpp

```diff
@@ -3197,7 +3198,7 @@ namespace {
                                            TVO_CanBindToNoEscape);
 
       Type optTy = getOptionalType(expr->getSubExpr()->getLoc(), valueTy);
-      if (!optTy)
+      if (!optTy || optTy->hasError())
```

Comment:
This should never be null

Reply:
I wonder why I left the two checks in. Probably an oversight but I will check whether I had noticed some case where optTy could be nullptr.